src/test/kotlin/cacophony/controlflow/FunctionHandlerTest.kt

```diff
@@ -728,34 +725,101 @@ class FunctionHandlerTest {
                 )
             val gHandler = FunctionHandlerImpl(gDef, gAnalyzed, listOf())
             val fHandler = FunctionHandlerImpl(fDef, fAnalyzed, listOf(gHandler))
-            gHandler.registerVariableAllocation(
-                gHandler.getStaticLink(),
-                VariableAllocation.OnStack(48),
-            )
-            fHandler.registerVariableAllocation(
-                fHandler.getStaticLink(),
-                VariableAllocation.OnStack(16),
-            )
 
             // when
             val staticLinkAccess = fHandler.generateVariableAccess(gHandler.getStaticLink())
 
             // then
             assertThat(staticLinkAccess).isEqualTo(
-                CFGNode.MemoryAccess( // [[rbp + 16] + 48]
+                CFGNode.MemoryAccess(
+                    // [[rbp] + 0]
                     CFGNode.Addition(
                         CFGNode.MemoryAccess(
-                            CFGNode.Addition(
-                                CFGNode.RegisterUse(Register.FixedRegister(X64Register.RBP)),
-                                CFGNode.Constant(16),
-                            ),
+                            CFGNode.RegisterUse(Register.FixedRegister(X64Register.RBP)),
                         ),
-                        CFGNode.Constant(48),
+                        CFGNode.Constant(0),
                     ),
                 ),
             )
         }
 
+        @Test
+        fun `calls its sibling with static link to parent `() {
+            // let g = [] -> Int => (
+            //   let h = [] -> Int => 42
+            //   let f = [] -> Int => h[]  # should pass a static link to parent
+            // );
+
+            // given
+            val hDef =
+                Definition.FunctionDeclaration(
+                    mockRange,
+                    "h",
+                    null,
+                    listOf(),
+                    Type.Basic(mockRange, "Int"),
+                    Literal.IntLiteral(mockRange, 42),
+                )
+            val fDef =
+                Definition.FunctionDeclaration(
+                    mockRange,
+                    "f",
+                    null,
+                    listOf(),
+                    Type.Basic(mockRange, "Int"),
+                    FunctionCall(mockRange, VariableUse(mockRange, "h"), listOf()),
+                )
+            val gDef =
+                Definition.FunctionDeclaration(
+                    mockRange,
+                    "g",
+                    null,
+                    listOf(),
+                    Type.Basic(mockRange, "Int"),
+                    Block(mockRange, listOf(hDef, fDef)),
+                )
+            val hAnalyzed =
+                AnalyzedFunction(
+                    ParentLink(gDef, true),
+                    setOf(),
```

Comment:
Similarly as above, I think that `emptySet()` would be more idiomatic thank `setOf()`. Could you please change the occurrences in this file?